### PR TITLE
libinit: Implement ErrNoModulesFound

### PR DESCRIPTION
**Pull request by Foxboron**
While rewriting stuff in https://github.com/u-root/u-root/pull/2482 I apparently overlooked the fact that we'd abort if any the functions fail. `InstallModulesFromDir` is always going to return an error unless there is `/lib/modules/*.ko` which means we won't be looking at the other places.

A quickfix is to just never return an error on module loading. Another option would be to make the error from `InstallModulesFromDir` more granular so we can differentiate between "we found no files in this directory" and "we failed to load modules from this directory".

I'm unsure what people would prefer?

### CODE REVIEW

**Review by 10000TB — DISMISSED** [review]

**Comment by 10000TB** [comment]
I think you could log errors in a more readable way, 

e.g. so if modules files are not present, it proceeds to find conf,     
etc, and may eventually successfully load some modules. 

In that case, you still print an error from not finding ko files.

File: pkg/libinit/root_linux.go [hunk 1]
```diff
@@ -271,26 +271,25 @@ func NewInitModuleLoader() *InitModuleLoader {
 // Useful for modules that need to be loaded for boot (ie a network
 // driver needed for netboot). It skips over blacklisted modules in
 // excludedMods.
-func InstallAllModules() error {
+func InstallAllModules() {
```
Comment (abrender) [inline pkg/libinit/root_linux.go:274]: May I suggest that this function signature not change and that it continue to return errors?  You can still ignore the returned errors in init_linux.go (and please add a comment into the code about why you're ignore the errors).

This way InstallAllModules() can be used by other code that might not want to ignore errors.

Thanks!

Comment (Foxboron) [reply pkg/libinit/root_linux.go:274]: Thats why I'm wondering if we should have a `ErrNoModulesFound` and a normal error. Then we can differentiate between the errors that gets returned.

Comment (10000TB) [reply pkg/libinit/root_linux.go:274]: That sounds good. But if you return the error, how could you proceed further ?

Comment (10000TB) [reply pkg/libinit/root_linux.go:274]: or maintain a list of errors ? -- if run into an error, append it, and in the end return it anyway.

**Comment by rminnich** [comment]
Things are about to get better in  error handling. 
https://github.com/golang/go/issues/53435

File: pkg/libinit/root_linux.go (hunk 1 above)
Comment (rminnich) [reply pkg/libinit/root_linux.go:274]: let's wait until go 1.20 for a list of errors, that stuff all gets better soon. 

And let's make our own ErrNoModulesFound for now?

**Comment by Foxboron** [comment]
Changed the implementation to implement `ErrNoModulesFound`. This also removes the annoying startup message :)

**Review by rminnich — APPROVED** [review]

**Pull request merged**